core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -2332,12 +2330,24 @@ String memberId() {
             return memberId;
         }
 
+        private RecordState rollbackState() {
+            if (rollbackState == null) {
+                // This case could occur when the batch/offset hasn't transitioned even once or the state transitions have
+                // been committed.
+                return null;
+            }
+            return rollbackState.state;
+        }
+
         // Visible for testing.
         TimerTask acquisitionLockTimeoutTask() {
             return acquisitionLockTimeoutTask;
         }
 
-        void updateAcquisitionLockTimeoutTask(AcquisitionLockTimerTask acquisitionLockTimeoutTask) {
+        void updateAcquisitionLockTimeoutTask(AcquisitionLockTimerTask acquisitionLockTimeoutTask) throws IllegalStateException {
```

Comment:
Is it really a IllesgalStateException or IllegalArgumentException for the method?